senders/formatter.go

```diff
@@ -308,3 +314,52 @@ func isUUIDFormat(str string) bool {
 	}
 	return true
 }
+
+func sanitizeInternal(str string) string {
+	sb := GetBuffer()
+	defer PutBuffer(sb)
+
+	skipHead := 0
+	if strings.HasPrefix(str, deltaPrefix) {
+		sb.WriteString(deltaPrefix)
+		skipHead = 3
+	}
+	if strings.HasPrefix(str, altDeltaPrefix) {
+		sb.WriteString(altDeltaPrefix)
+		skipHead = 2
+	}
+	if str[0] == 126 {
+		sb.WriteString(string(str[0]))
+		skipHead = 1
+	}
+
+	for i := 0; i < len(str); i++ {
+		if skipHead > 0 {
+			i += skipHead
+			skipHead = 0
+		}
+		cur := str[i]
+		strCur := string(cur)
+		isLegal := true
+
+		if !(44 <= cur && cur <= 57) && !(65 <= cur && cur <= 90) && !(97 <= cur && cur <= 122) && cur != 95 {
+			isLegal = false
+		}
+		if isLegal {
+			sb.WriteString(strCur)
+		} else {
+			sb.WriteString("-")
+		}
+	}
+	return sb.String()
+}
+
+func sanitizeValue(str string) string {
+	res := strings.TrimSpace(str)
+	if strings.Contains(str, "\"") || strings.Contains(str, "'") {
+		resReg, _ := regexp.Compile("\"")
```

Comment:
Make these global vars and pre-compile the regular expression. You can then reuse it within this method. Compiling it for every call is expensive and should be avoided.

Use `MustCompile(...)` for initializing the global vars.

Reply:
these two can actually be const. made them const